src/Icicle/Avalanche/Statement/Simp/Melt.hs

```diff
@@ -185,6 +201,12 @@ meltAccumulators a_fresh statements
                     . Read nb' nb avt tb
                     $ ss'
 
+           | Just (Mutable, StructT ts, nas)            <- Map.lookup acc env'
+           , fts                                        <- Map.elems (getStructType ts)
+           -> do ns' <- replicateM (length fts) (freshPrefix' n)
+                 ss' <- substXinS a_fresh n (primMkStruct ts ns') ss
+                 go $ foldr (\(n',na,t) -> Read n' na avt t) ss' (List.zip3 ns' nas fts)
```

Comment:
Yeah I was surprised! I thought this would be much harder, but it's almost the nicest of all.